### PR TITLE
[CI] Run MacOS build job only on main, release branches and omnibus changes

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR makes MacOS build job only run on `main`, release branches and `omnibus/` changes.

### Motivation

Reduce the MacOS runners load to avoid running out of runners on Github Actions.

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]

File: .gitlab/package_build/dmg.yml [hunk 1]
```diff
@@ -23,7 +23,9 @@
 agent_dmg-x64-a7:
   extends: .agent_build_common_dmg
   rules:
-    - !reference [.on_a7]
+    - !reference [.on_omnibus_change]
```
Comment (KSerrania) [inline .gitlab/package_build/dmg.yml:26]: I think we should update the "on omnibus change" rule, it is too restrictive (it only triggers on changes to `omnibus/`). For instance, it doesn't trigger on Gitlab CI config changes, which do affect packaging.

I think this should be renamed to `.on_packaging_changes`, and at least include changes to `.gitlab-ci.yml`, `.gitlab/package_build.yml`, `release.json`, and files in `.gitlab/package_build`.

Comment (amenasria) [reply .gitlab/package_build/dmg.yml:26]: Good point, on it.

Comment (amenasria) [reply .gitlab/package_build/dmg.yml]: Fixed with [e91d0b0](https://github.com/DataDog/datadog-agent/pull/22882/commits/e91d0b0f2b1758cf1fa59031f9133cb033de9205) !

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1143,11 +1143,15 @@ workflow:
...
-.on_omnibus_change:
+.on_packaging_change:
   - !reference [.except_mergequeue] # The prerequisites are not run in the mergequeue pipeline so we need to skip this rule
   - changes:
       paths:
         - omnibus/**/*
+        - .gitlab-ci.yml
+        - .gitlab/package_build.yml
+        - release.json
+        - .gitlab/package_build
```
Comment (KSerrania) [inline .gitlab-ci.yml:1154]:
```suggestion
        - .gitlab/package_build/**/*
```
?

(I'm assuming that is needed, given how the first rule is written, but I haven't checked)

Comment (amenasria) [reply .gitlab-ci.yml:1154]: You're right, thanks !

**Pull request merged**